docker/build.dockerfile

```diff
@@ -0,0 +1,11 @@
+ARG MONTAGU_GIT_ID="UNKNOWN"
```

Comment:
this is mostly just copied from previous location but github hasn't worked that out. the change is to the FROM bit - we now use a tagged image from docker hub because it gets built in a previous step and buildkite doesn't guarantee continuity of agents between steps (previously would just build a local image, call it "montagu-portal-shared-build-env" and inherit from that in the next step.)